modules/core/04-channel/keeper/upgrade.go

```diff
@@ -626,9 +627,20 @@ func (k Keeper) ChanUpgradeCancel(ctx sdk.Context, portID, channelID string, err
 		return errorsmod.Wrap(commitmenttypes.ErrInvalidProof, "cannot submit an empty error receipt proof unless the sender is authorized to cancel upgrades AND channel is not in FLUSHCOMPLETE")
 	}
 
-	// the error receipt should also have a sequence greater than or equal to the current upgrade sequence.
-	if errorReceipt.Sequence < channel.UpgradeSequence {
-		return errorsmod.Wrapf(types.ErrInvalidUpgradeSequence, "error receipt sequence (%d) must be greater than or equal to current upgrade sequence (%d)", errorReceipt.Sequence, channel.UpgradeSequence)
+	// REPLAY PROTECTION: the error receipt should also have a sequence greater than or equal to the current upgrade sequence,
+	// except when the channel state is FLUSHCOMPLETE, in which case the sequences must match. This is required
+	// to guarantee that this channel end does not cancel the upgrade in case the counterparty has completed
+	// the upgrade and moved to OPEN, initiated a new upgrade (and thus incremented the upgrade sequence) and
+	// cancelled, all in the same block. Otherwise, the channel would open, but one end would have upgraded and
+	// the other not.
```

Comment:
```suggestion
	// REPLAY PROTECTION: The error receipt MUST have a sequence greater than or equal to the current upgrade sequence,
	// except when the channel state is FLUSHCOMPLETE, in which case the sequences MUST match. This is required
	// to guarantee that when a counterparty successfully completes an upgrade and moves to OPEN, this channel
	// cannot cancel its upgrade.  Without this strict sequence check, it would be possible for the counterparty
	// to complete its upgrade, move to OPEN, initiate a new upgrade (and thus increment the upgrade sequence) and
	// then cancel the new upgrade, all in the same block. This results in a valid error receipt being written at channel.UpgradeSequence + 1. 
	// The desired behaviour in this circumstance is for this channel to complete its current upgrade despite proof
	// of an error receipt at a greater upgrade sequence
```